src/deploy/AWSDeployer.js

```diff
@@ -691,6 +727,68 @@ export default class AWSDeployer extends BaseDeployer {
     if (cleanup) {
       await this.cleanUpIntegrations(functionName);
     }
+
+    await this.updateAuthorizers(ApiId, functionName, aliasArn);
+  }
+
+  async updateAuthorizers(ApiId, functionName, aliasArn) {
+    const cfg = this._cfg;
+    if (!cfg.createAuthorizer) {
+      return;
+    }
+
+    const AUTH_URI_PREFIX = `arn:aws:apigateway:${cfg.region}:lambda:path/2015-03-31/functions/`;
+    const accountId = aliasArn.split(':')[4];
+    this.log.info(chalk`--: patching authorizers...`);
+    const authorizers = await this.fetchAuthorizers(ApiId);
+    const versions = this.getLinkVersions();
+    for (const version of versions) {
+      const props = {
+        ...this.cfg,
+        ...this.cfg.properties,
+        // overwrite version with link name
+        version,
+      };
+      const authorizerName = ActionBuilder.substitute(cfg.createAuthorizer, props).replace(/\./g, '_');
+      const existing = authorizers.find((info) => info.Name === authorizerName) || {};
+      let { AuthorizerId } = existing;
+      if (AuthorizerId) {
+        const res = await this._api.send(new UpdateAuthorizerCommand({
+          ApiId,
+          AuthorizerId,
+          AuthorizerUri: `${AUTH_URI_PREFIX}${aliasArn}/invocations`,
+        }));
+        this.log.info(chalk`{green ok}: updated authorizer: {blue ${res.Name}}`);
+      } else {
+        const res = await this._api.send(new CreateAuthorizerCommand({
+          ApiId,
+          AuthorizerPayloadFormatVersion: '2.0',
+          AuthorizerType: 'REQUEST',
+          AuthorizerUri: `${AUTH_URI_PREFIX}${aliasArn}/invocations`,
+          AuthorizerResultTtlInSeconds: 0,
+          EnableSimpleResponses: true,
+          IdentitySource: ['$request.header.Authorization'],
+          Name: authorizerName,
+        }));
+        AuthorizerId = res.AuthorizerId;
+        this.log.info(chalk`{green ok}: created authorizer: {blue ${res.Name}}`);
+      }
+
+      // add permission to alias for the API Gateway is allowed to invoke the authorized function
+      try {
+        const sourceArn = `arn:aws:execute-api:${this._cfg.region}:${accountId}:${ApiId}/authorizers/${AuthorizerId}`;
+        await this._lambda.send(new AddPermissionCommand({
+          FunctionName: aliasArn,
+          Action: 'lambda:InvokeFunction',
+          SourceArn: sourceArn,
+          Principal: 'apigateway.amazonaws.com',
+          StatementId: crypto.createHash('md5').update(aliasArn + sourceArn).digest('hex'),
```

Comment:
```suggestion
          StatementId: crypto.createHash('sha256').update(aliasArn + sourceArn).digest('hex'),
```
to avoid the `1 for Use of a broken or weak cryptographic algorithm` alert